md-rest-api/src/main/java/com/wipro/ats/bdre/md/rest/AppDeploymentQueueAPI.java

```diff
@@ -46,9 +42,182 @@
+            LOGGER.info("app deployment queue Id is "+adqId);
+             returnedAppDeploymentQueue.setProcessId(appDeploymentQueue.getProcessId());
+             returnedAppDeploymentQueue.setAppName(appDeploymentQueue.getAppName());
+             returnedAppDeploymentQueue.setAppDomain(appDeploymentQueue.getAppDomain());
+             returnedAppDeploymentQueue.setUsername(principal.getName());
+             returnedAppDeploymentQueue.setAppDeploymentQueueId(adqId);
+             returnedAppDeploymentQueue.setAppDeploymentStatusId((short) 0);
+             restWrapper = new RestWrapper(returnedAppDeploymentQueue, RestWrapper.OK);
+        } catch (MetadataException e) {
+            LOGGER.error(e);
+            restWrapper = new RestWrapper(e.getMessage(), RestWrapper.ERROR);
+        }
+        return restWrapper;
+    }
+
+
+    @RequestMapping(value = "/merge/{id}", method = RequestMethod.POST)
+    @ResponseBody
+    public
+    RestWrapper merge(@PathVariable("id") Long queueId, Principal principal) {
+        RestWrapper restWrapper = null;
+        LOGGER.info("queue id is "+queueId);
+        LOGGER.info("user name is "+principal.getName());
+        AppDeploymentQueue returnedAppDeploymentQueue=new AppDeploymentQueue();
+        try{
+            String temp;
+            BufferedReader br = null;
+            String jsonfile="";
+            String homeDir = System.getProperty("user.home");
+            LOGGER.info("home directory" + homeDir);
+            br = new BufferedReader(new FileReader(homeDir+"/bdreappstore/store.json"));
+            while ((temp=br.readLine()) != null) {
+                jsonfile=jsonfile+temp;
+            }
+            ObjectMapper mapper = new ObjectMapper();
+            mapper.configure(DeserializationConfig.Feature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+            AppStore appStore = mapper.readValue(jsonfile, AppStore.class);
+            LOGGER.info("size of app catagory is "+appStore.getApplicationList().size());
+            com.wipro.ats.bdre.md.dao.jpa.AppDeploymentQueue appDeploymentQueue=appDeploymentQueueDAO.get(queueId);
+            LOGGER.info("AppDeploymentQueue  is "+appDeploymentQueue);
+            for (StoreJson storeJson:appStore.getApplicationList())
+            {
+                if (storeJson.getId().equals(appDeploymentQueue.getAppDomain()))
+                {
+                    LOGGER.info("no of apps in this catagory without updating is "+storeJson.getColumns().size());
+                    int alreadyExistedApp=0;
+                    AppValues addedApp=new AppValues();
+                    for (AppValues appValues:storeJson.getColumns())
+                    {
+                        if (appValues.getName().equals(appDeploymentQueue.getAppName()))
+                        {
+                            alreadyExistedApp=1;
+                        }
+                    }
+
+                    if (alreadyExistedApp==0)
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS134)